larray/core.py

```diff
@@ -8429,7 +8443,7 @@ def diag(a, k=0, axes=(0, 1), ndim=2, split=True):
                     for name, labels in zip(axes_names, axes_labels)]
         else:
             axes = [axis] + [axis.copy() for _ in range(ndim - 1)]
-        res = zeros(axes, dtype=a.dtype)
+        res = zeros(axes, title=a.title, dtype=a.dtype)
```

Comment:
There is another else below in which we return a.points[indexer]. I think the title will remain in this case.